lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -1209,3 +1204,188 @@ def get_course_run_status(course_overview, certificate_info):
+        * 403: The requesting user lacks access for the given program/course.
+        * 404: The requested program or course does not exist.
+
+    **Response**
+
+        In the case of a 200/207/422 response code, the response will include a
+        paginated data set.  The `results` section of the response consists of a
+        list of grade records, where each successfully loaded record contains:
+          * student_key: The identifier of the student enrolled in the program and course.
+          * letter_grade: A letter grade as defined in grading policy
+            (e.g. 'A' 'B' 'C' for 6.002x) or None.
+          * passed: Boolean representing whether the course has been
+            passed according to the course's grading policy.
+          * percent: A float representing the overall grade for the course.
+        and failed-to-load records contain:
+          * student_key
+          * error: error message from grades Exception
+
+    **Example**
+
+        207 Multi-Status
+        {
+            "next": null,
+            "previous": "http://example.com/api/program_enrollments/v1/programs/{program_uuid}/courses/{course_id}/grades/?cursor=abcd",
+            "results": [;
+                {
+                    "student_key": "01709bffeae2807b6a7317",
+                    "letter_grade": "Pass",
+                    "percent": 0.95,
+                    "passed": true
+                },
+                {
+                    "student_key": "2cfe15e3380a52e7198237",
+                    "error": "Timeout while calculating grade"
+                },
+                ...
+            ],
+        }
+
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        OAuth2AuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (permissions.JWT_RESTRICTED_APPLICATION_OR_USER_ACCESS,)
+    pagination_class = ProgramEnrollmentPagination
+
+    @verify_course_exists
+    @verify_program_exists
+    def get(self, request, program_uuid=None, course_id=None):
+        """
+        Defines the GET list endpoint for ProgramCourseGrade objects.
+        """
+        course_key = CourseKey.from_string(course_id)
+        grade_results = self._load_grade_results(program_uuid, course_key)
+        serializer = ProgramCourseGradeResultSerializer(grade_results, many=True)
+        response_code = self._calc_response_code(grade_results)
+        return self.get_paginated_response(serializer.data, status_code=response_code)
+
+    def _load_grade_results(self, program_uuid, course_key):
+        """
+        Load grades (or grading errors) for a given program courserun.
+
+        Arguments:
+            program_uuid (str)
+            course_key (CourseKey)
+
+        Returns: list[ProgramCourseGradeResult|ProgramCourseGradeErrorResult]
+        """
+        enrollments_qs = use_read_replica_if_available(
+            ProgramCourseEnrollment.objects.filter(
```

Comment:
For now, it would return the enrollments linked to a persisted `ProgramCourseEnrollment`. What about those Verified enrollments to a course run that's part of a linked MM program? Should we also return grades for those course runs?

Reply:
Isn't it possible that there are verified students in one of those MM courses that aren't really part of the program?

Reply:
That's a good point. I guess we should clarify with Product.

Reply:
The ticket is pretty explicit that this response won't include MM courses, etc. But yeah clarifying with product is probably best.

Reply:
Oh, yeah, that's a good point.

Reply:
Talked to Deen about this when creating the ticket, the only grades we should be returning are for enrollments explicitly enrolled by the partner.